presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -153,6 +153,15 @@ public long getSizeInMemory()
         return sizeInMemory;
     }
 
+    /**
+     * building hash sorted results requires memory for sorting group IDs.
+     * This method returns size of that memory requirement.
+     */
+    public long getGroupIdsSortingSize()
+    {
+        return getGroupCount() * Integer.BYTES;
```

Comment:
Ops, indeed. Sorry, old habits from C++ world. I'm really surprised how tricky it is to actually sort arrays of int in Java... I can not use int[] because of GC AND because there is no `Arrays.sort(int[]) `with a comparator. `IntArrayList` for sorting with comparator also creates temporary `Integer[]`. Same as `IntStream.sorted()`. `IntBigArray` on the other hand has no sort method at all.

I have implemented `List<Integer>` wrapper around `IntBigArray`.

I don't want to pre allocate this array, because it would be unused with disabled spilling.